updated travis.yml file

alixdamman opened the pull request:
- clone larray from github repository and install it manually (larray-eurostat and larray are developed in parallel)
- replaced nose by pytest

alixdamman committed:
travisCI:
- removed python 3.4
- clone larray from github repository and install it manually (larray-eurostat and larray are developed in parallel)
- replaced nose by pytest

pytest:
- added setup.cfg file

gdementen commented on .travis.yml:
```diff
@@ -41,16 +40,21 @@ before_install:
     # - sudo ln -s /run/shm /dev/shm
 
+    # clone larray repository
+    - git clone https://github.com/liam2/larray.git larray
```
don't forget to set it to larray-project if you merge this after liam2/larray#600

alixdamman replied:
Yes, I know. 
I wanted to run travis first because the few doctests present in larray-eurostat crash on my computer. 
For some reasons, travis didn't start?

gdementen replied:
Travis CI was not configured to build PR for this project. I just changed that.

alixdamman replied:
Thanks

gdementen reviewed (approved):
LGTM

alixdamman committed:
- updated doctest of eurostat_get() function (previous Eurostat data seems no longer available).
- added a try/catch in eurostat_get() function
- updated unittest test_eurostat_get()

gdementen commented on larray_eurostat/tsv.py:
```diff
@@ -50,19 +50,23 @@ def eurostat_get(indicator, drop_markers=True):
+                    # strip markers except on first line
+                    s = s[:first_line_end] + remove_chars(s[first_line_end:], ' dbefcuipsrzn:')
+                return read_eurostat(StringIO(s))
+            except:
```
this kind of blanket exception is usually not a good idea. There are several different things which can go wrong in this code and this could make debugging harder. If some particular exception is awful to understand, you could catch just that exception and "translate" it to something more intelligible.

gdementen commented on larray_eurostat/tsv.py:
```diff
@@ -50,19 +50,23 @@ def eurostat_get(indicator, drop_markers=True):
+                    s = s[:first_line_end] + remove_chars(s[first_line_end:], ' dbefcuipsrzn:')
+                return read_eurostat(StringIO(s))
+            except:
+                print("Can't open file {}".format(f.geturl()))
```
This is not good. We must throw an exception, otherwise the function will happily return None and the user code will crash at an unrelated point later

gdementen commented on larray_eurostat/tsv.py:
```diff
@@ -50,19 +50,24 @@ def eurostat_get(indicator, drop_markers=True):
+                    s = s[:first_line_end] + remove_chars(s[first_line_end:], ' dbefcuipsrzn:')
+                return read_eurostat(StringIO(s))
+            except Exception as e:
+                e.args = (e.args[0] + " \nCan't open file {}".format(f.geturl()),)
```
It would be nice to have a test case for the failing case.

alixdamman commented:
Still OK?

gdementen reviewed (approved):
LGTM
